tensorflow_datasets/image/lost_and_found.py

```diff
@@ -0,0 +1,223 @@
+          segmentation_labels=True,
+          instance_ids=False,
+          disparity_maps=False,
+          use_16bit=False,
+      ),
+      LostAndFoundConfig(
+          name='stereo_disparity',
+          description='Lost and Found stereo images and disparity maps.',
+          version="1.0.0",
+          right_images=True,
+          segmentation_labels=False,
+          instance_ids=False,
+          disparity_maps=True,
+          use_16bit=False,
+      ),
+      LostAndFoundConfig(
+          name='full',
+          description='Full Lost and Found dataset.',
+          version="1.0.0",
+          right_images=True,
+          segmentation_labels=True,
+          instance_ids=True,
+          disparity_maps=True,
+          use_16bit=False,
+      ),
+      LostAndFoundConfig(
+          name='full_16bit',
+          description='Full Lost and Found dataset.',
+          version="1.0.0",
+          right_images=True,
+          segmentation_labels=True,
+          instance_ids=True,
+          disparity_maps=True,
+          use_16bit=True,
+      )]
+
+  def _info(self):
+    possible_features = {
+        'image_left': tfds.features.Image(shape=(1024, 2048, 3),
+                                          encoding_format='png'),
+        'image_right': tfds.features.Image(shape=(1024, 2048, 3),
+                                           encoding_format='png'),
+        'segmentation_label': tfds.features.Image(shape=(1024, 2048, 1),
+                                                  encoding_format='png'),
+        'instance_id': tfds.features.Image(shape=(1024, 2048, 1),
+                                           encoding_format='png'),
+        'disparity_map': tfds.features.Image(shape=(1024, 2048, 1),
+                                             encoding_format='png')}
+    return tfds.core.DatasetInfo(
+        builder=self,
+        # This is the description that will appear on the datasets page.
+        description=_DESCRIPTION,
+        # tfds.features.FeatureConnectors
+        features=tfds.features.FeaturesDict({
+            'image_id': tfds.features.Text(),
+            **{feat: possible_features[feat]
+               for feat in self.builder_config.features}}),
+        # Homepage of the dataset for documentation
+        urls=['http://www.6d-vision.com/lostandfounddataset'],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    base_url = 'http://www.dhbw-stuttgart.de/~sgehrig/lostAndFoundDataset/{}.zip'
+    download_urls = {
+        'image_left': base_url.format(self.builder_config.left_image_string)}
+    if 'image_right' in self.builder_config.features:
+      download_urls['image_right'] = base_url.format(
+          self.builder_config.right_image_string)
+    if 'segmentation_label' in self.builder_config.features \
+            or 'instance_id' in self.builder_config.features:
+      download_urls['gt'] = base_url.format('gtCoarse')
```

Comment:
AFAIK - you can pass the same URL multiple times to download_and_extract (and will do the "smart" thing).

this might make your code easier (as you can pass download_urls['segmentation_label'] and download_urls['instance_id'])

Reply:
I will update this, your suggestions is much more elegant and also better readable, thanks!